pkg/disttask/framework/taskexecutor/task_executor.go

```diff
@@ -548,74 +535,27 @@ func (e *BaseTaskExecutor) refreshTask() error {
 }
 
 func (e *BaseTaskExecutor) registerRunStepCancelFunc(cancel context.CancelCauseFunc) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	e.mu.runtimeCancel = cancel
+	e.runtimeCancel = cancel
 }
 
 func (e *BaseTaskExecutor) unregisterRunStepCancelFunc() {
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	e.mu.runtimeCancel = nil
+	e.runtimeCancel = nil
 }
 
 func (e *BaseTaskExecutor) cancelRunStepWith(cause error) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
-	if e.mu.runtimeCancel != nil {
-		e.mu.runtimeCancel(cause)
-	}
-}
-
-func (e *BaseTaskExecutor) onError(err error) {
-	if err == nil {
-		return
+	if e.runtimeCancel != nil {
+		e.runtimeCancel(cause)
```

Comment:
We have two places call `cancelRunStepWith`, and only one place that fits `e.runtimeCancel != nil`. So could we remove `cancelRunStepWith`.